avocado/core/task/statemachine.py

```diff
@@ -79,14 +79,23 @@ async def triage(self):
         except IndexError:
             return
 
-        requirements_ok = await self._spawner.check_task_requirements(runtime_task)
-        if requirements_ok:
-            async with self._state_machine.lock:
-                self._state_machine.ready.append(runtime_task)
-        else:
-            async with self._state_machine.lock:
-                self._state_machine.finished.append(runtime_task)
-                runtime_task.status = 'FAILED ON TRIAGE'
+        # right now, support triage based on requirements only
+        if runtime_task.task.prereqs:
```

Comment:
If `prereqs` could return always a list, this If would not be necessary.

Reply:
If the list is empty this will execute as if it is `None` so this is probably correct implementation.

Reply:
You are right, @pevogam, I forgot about line 84 here.

I would probably change `_state_machine.finished` to a dict (using the `task.uid` as keys), instead of a list, to avoid this dict comprehension and to do everything inside the loop.

But, probably this would need a much bigger refactoring. So yes, I'm ok with that.

Reply:
That makes sense, and I think `prereqs` is already always a list. I'll check that.